doc/commands/build.rst

```diff
@@ -97,13 +97,67 @@ options
     wildcards (``**``), extended globbing (``@(…)``, ``+(…)``, etc.),
     and negation (``!…``).
 
+    Patterns should be relative to the build directory.
```

Comment:
_Potentially tangential to this PR_

Reading this code my understanding is that only the build directory is uploaded, and so patterns are relative to that. How does this work when the build directory is (e.g.) `ingest/` but within that the Snakefile refers back to a parent directory?

Reply:
Ah, that's a great point. The directory here that matters—the one that's uploaded and one that patterns must be relative to—is influenced by the `nextstrain-pathogen.yaml` file. For example, running `nextstrain build ingest/` in a directory with `nextstrain-pathogen.yaml` will require patterns be relative to the directory containing the marker file, not the `ingest/` directory, i.e. to exclude something in `ingest/` would require the pattern to be `ingest/x/y/z` not just `x/y/z`.

I'll update the docs here.

Reply:
Actually, I want to leave the option docs/help as-is for now for this PR. I'm realizing there's larger documentation reworking to do to clarify terminology here as a result of https://github.com/nextstrain/cli/pull/355 (something I'd postponed in that PR). Writing a clear help snippet for these pattern options will be much easier and less inconsistent/confusing once the larger terminology shift and doc updates happen. I should prioritize that.

Reply:
https://github.com/nextstrain/cli/issues/371

Reply:
👍  totally forgot about this subtlety. Is this behaviour mirrored in what directories are bind-mounted into the docker container?

Reply:
Yep.

Before #355 the directory given to `nextstrain build` ended up as `/nextstrain/build` in the container and `/nextstrain/build` was the fixed working directory for the container process.

After #355, the `/nextstrain/build` mount is either the directory given to `nextstrain build` or, if present, the closest ancestor of the given directory which contains a `nextstrain-pathogen.yaml`. The working directory for the container process is the `/nextstrain/build` path corresponding to the directory given to `nextstrain build`.